tests/functional/test_textx_language.py

```diff
@@ -108,6 +108,92 @@ def test_unreferenced_abstract_rule():
     assert mm['Third']._tx_type == RULE_COMMON
 
 
+def test_abstract_rule_with_multiple_match_rule_refs():
+    """
+    Test that a single alternative of abstract rule can reference multiple
+    match rules
+    """
+    grammar = """
+    Rule: STRING|Rule1|ID|'#' INT '--';
+    Rule1: a='a'; // common rule
+    Prefix: '#';
+    Sufix: '--';
+    """
+    meta = metamodel_from_str(grammar)
+    model = meta.model_from_str('# 23 --')
+    assert meta['Rule']._tx_type is RULE_ABSTRACT
+    assert model == '#23--'
+
+
+def test_abstract_rule_with_multiple_rule_refs():
+    """
+    Test that a single alternative of abstract rule can reference multiple
+    match rules with a single common rule.
+    """
+    grammar = """
+    Rule: STRING|Rule1|ID|Prefix Rule1 Sufix;
+    Rule1: a=INT; // common rule
+    Prefix: '#';
+    Sufix: '--';
+    """
+    meta = metamodel_from_str(grammar)
+    model = meta.model_from_str('# 23 --')
+    assert meta['Rule']._tx_type is RULE_ABSTRACT
+    assert model.a == 23
+
+
+def test_abstract_rule_with_multiple_common_rule_refs():
+    """
+    Test that a single alternative of abstract rule can not reference multiple
+    common rules.
+    """
+    grammar = """
+    Rule: STRING|Rule1|ID|Prefix Rule1 Sufix Rule2;  // Reference both Rule1 and Rule2
```

Comment:
Good example for "If there are multiple common rule references the result will be the first in the order of definition".

Reply:
+1. Added to the docs.